### ISSUE TITLE
Error while creating terraform plan with RDS dependency 

### ISSUE THREAD

**Issue by manjeetvk** [ariga/terraform-provider-atlas#20]
looks like `atlas_schema` needs a valid connection URL. We are getting the below error in terraform plan as the RDS cluster is not present. Below is the terraform plan:

**Script:**

```
resource "aws_rds_cluster" "db" {
  apply_immediately               = true
  cluster_identifier              = local.db_namespace
  engine                          = "aurora-mysql"
 ......
}
resource "atlas_schema" "test_db" {
  hcl = file("${path.module}/db-schema/schema.hcl")
  url = "mysql://${aws_rds_cluster.db.master_username}:${aws_rds_cluster.db.master_password}@${aws_rds_cluster.db.endpoint}:${aws_rds_cluster.db.port}/${aws_rds_cluster.db.database_name}"
  depends_on = [aws_rds_cluster.db]
}

```

**ERROR:**

`Error: sql/sqlclient: no opener was register with name ""`

**Comment by manjeetvk** [comment]
looks like the provider needs a valid URL during terraform plan state.

**Comment by arohter** [comment]
`aws_rds_cluster.db.endpoint` would only be available at *apply* time (initial creation), but it looks like Atlas is trying to fully resolve the url at _plan_ time, which won't work for dependent resources that need to be created first.

**Comment by arohter** [comment]
Confirmed bug in provider, trying to fully result url during plan phase:
```
resource "aws_db_instance" "db" {
  allocated_storage    = 10
  engine               = "mysql"
  engine_version       = "5.7"
  instance_class       = "db.t3.micro"
  db_name              = "mydb"
  username             = "foo"
  password             = "foobarbaz"
  parameter_group_name = "default.mysql5.7"
  skip_final_snapshot  = true
}


resource "atlas_schema" "table" {
  hcl = file("my-table.hcl")
  url = "mysql://${aws_db_instance.db.username}:${aws_db_instance.db.password}@${aws_db_instance.db.endpoint}:${aws_db_instance.db.port}/${aws_db_instance.db.db_name}"
  depends_on = [aws_db_instance.db]
}
```

```
│ Error: sql/sqlclient: no opener was register with name ""
│ 
│   with atlas_schema.table,
│   on main.tf line 20, in resource "atlas_schema" "table":
│   20: resource "atlas_schema" "table" {
```

**Comment by arohter** [comment]
Confirmed fixed by #21  👍

**Comment by a8m** [comment]
Fixed with #21 

Thank you @arohter and @manjeetvk.
